File: aoj/volume2/n0238/Main.java
import java.util.Scanner;

public class Main {

    public static void main(String[] args) {

        Scanner sc = new Scanner(System.in);

        while (true) {

            int objective = sc.nextInt();

            if (objective == 0) {
                break;
            }

            int numStudy = sc.nextInt();
            int studyTime = 0;

            for (int i = 0; i < numStudy; i++) {

                int startTime = sc.nextInt();
                int endTime = sc.nextInt();

                studyTime += endTime - startTime;

            }

            System.out.println(objective <= studyTime ? "OK" : (objective - studyTime));

        }

    }

}

File: aoj/volume2/n0238/Main.cpp
#include <iostream>

using namespace std;

int main() {

    while (true) {

        int objective;
        cin >> objective;

        if (objective == 0) {
            break;
        }

        int numStudy;
        cin >> numStudy;

        int studyTime = 0;

        for (int i = 0; i < numStudy; i++) {

            int startTime, endTime;
            cin >> startTime >> endTime;

            studyTime += endTime - startTime;

        }

        if (objective <= studyTime) {
            cout << "OK" << endl;
        } else {
            cout << (objective - studyTime) << endl;
        }

    }

    return 0;

}
